src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +387,160 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
                     manifest.Version = Manifest.SupportedVersions.Max().ToString();
                 }
 
-                manifest.AddLibrary(new LibraryInstallationState
+                LibraryInstallationState libraryInstallationState = new LibraryInstallationState
                 {
                     LibraryId = PackageId,
                     ProviderId = selectedPackage.ProviderId,
                     DestinationPath = InstallationFolder.DestinationFolder,
                     Files = SelectedFiles.ToList()
-                });
+                };
 
-                await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                await Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
                 EnvDTE.Project project = VsHelpers.DTE.SelectedItems.Item(1)?.ProjectItem?.ContainingProject;
                 project?.AddFileToProjectAsync(_configFileName);
 
-                await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
+                string configFilePath = Path.GetFullPath(_configFileName);
 
-                _dispatcher.Invoke(() =>
+                IVsTextBuffer document = rdt.FindDocument(configFilePath) as IVsTextBuffer;
+
+                // The file isn't open. So we'll write to disk directly
+                if (document == null)
                 {
-                    _closeDialog(true);
-                });
+                    manifest.AddLibrary(libraryInstallationState);
+
+                    await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+
+                    await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
```

Comment:
Does this have to be after awaits in this case, or could you just always do this before the if?

Reply:
Actually it can be outside. Moved it in iteration 10.

---
In reply to: [197931544](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197931544) [](ancestors = 197931544)